### PR TITLE
BA-25: first version of GuestCheckoutPage

### PR DESCRIPTION
**Pull request by ilku12**
test

### CODE REVIEW

File: src/main/java/engine/property/manager/PropertyManager.java [hunk 1]
```diff
@@ -44,6 +44,7 @@ public static void setProperty(String systemProperty, String systemStringValue)
 
     public static BrowserName getBrowserName() {
         return BrowserName.valueOf(getProperty(PropertyKeys.BROWSER_NAME));
+
```
Comment (Fishgame1) [inline src/main/java/engine/property/manager/PropertyManager.java:47]: Accidental change probably, to be removed:)

File: src/main/java/pages/GuestCheckoutPage.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    public By passwordInputLocator = By.id("AccountFrm_password");
+    public By passConfirmInputLocator = By.id("AccountFrm_confirm");
+    public By newsletterInputLocator = By.xpath("//*[@id='AccountFrm_newsletter1']/button");
+
+
+    public GuestCheckoutPage() {
+        this.basePage = new BasePage();
+    }
+
+    public void enterFirstName(String firstname) { basePage.enterTextToElement(firstnameInputLocator, firstname); }
```
Comment (Fishgame1) [inline src/main/java/pages/GuestCheckoutPage.java:35]: Use auto-formatting on all of the files:)

File: src/main/java/pages/GuestCheckoutPage.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        basePage.enterTextToElement(emailInputLocator, email);
+    }
+    public void enterTelephone(String telephone) { basePage.enterTextToElement(telephoneInputLocator, telephone); }
+    public void enterFax(String fax) {
+        basePage.enterTextToElement(faxInputLocator, fax);
+    }
+    public void enterCompany(String company) {
+        basePage.enterTextToElement(companyInputLocator, company);
+    }
+    public void enterAddress_1(String address_1) {
```
Comment (Fishgame1) [inline src/main/java/pages/GuestCheckoutPage.java:49]: enterFirstLineOfAddress? Or there are two separate adresses?

File: src/main/java/pages/GuestCheckoutPage.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        basePage.enterTextToElement(address1InputLocator, address_1);
+    }
+    public void enterAddress_2(String address_2) { basePage.enterTextToElement(address2InputLocator, address_2); }
+    public void enterCity(String city) {
+        basePage.enterTextToElement(cityInputLocator, city);
+    }
+
+    protected WebElement locateElement(By zoneIdInputLocator) {
```
Comment (Fishgame1) [inline src/main/java/pages/GuestCheckoutPage.java:57]: Why this is separate?

File: src/main/java/pages/GuestCheckoutPage.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        wait.until(ExpectedConditions.visibilityOfElementLocated(zoneIdInputLocator));
+        return getWebDriverInstance().findElement(zoneIdInputLocator);
+    }
+
+    public void enterPostcode(String postcode) { basePage.enterTextToElement(postcodeInputLocator, postcode); }
+
+    protected WebElement locateElement(By countryIdInputLocator) {
```
Comment (Fishgame1) [inline src/main/java/pages/GuestCheckoutPage.java:64]: Why this is as separate wait here?

File: src/main/java/pages/GuestCheckoutPage.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import static engine.drivers.WebDriverFactory.getWebDriverInstance;
+
+public class GuestCheckoutPage {
```
Comment (Fishgame1) [inline src/main/java/pages/GuestCheckoutPage.java:9]: Can you add Step class for that will use your methods to actually fill the page based on random customer (there is already method for getting him/her) 
?:)

File: src/test/java/temporary/tests/TestOpenPageChrome.java [hunk 7]
```diff
@@ -25,7 +25,8 @@ public void openPage() {
         String url = getProperty(BASE_URL);
         driver.navigate().to(url);
 
-        var welcomeMsg = By.xpath("//div[@class='welcome_msg']");
+        var welcomeMsg = By.xpath("//div[@clas" +
```
Comment (Fishgame1) [inline src/test/java/temporary/tests/TestOpenPageChrome.java:28]: Accidental change probably, to be reverted:) You can do it by using git checkout <filePath>

File: src/main/java/pages/GuestCheckoutPage.java (hunk 3 above)
Comment (dariabajda) [reply src/main/java/pages/GuestCheckoutPage.java:49]: also Java doesn't like kebab_case :) use camelCase for variable names

File: src/main/java/pages/GuestCheckoutPage.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    public By address1InputLocator = By.id("AccountFrm_address_1");
+    public By address2InputLocator = By.id("AccountFrm_address_2");
+    public By cityInputLocator = By.id("AccountFrm_city");
+    public By zoneIdInputLocator = By.xpath("//ul[contains(@class, 'currency')]/parent::*");
+    public By postcodeInputLocator = By.id("AccountFrm_postcode");
+    public By countryIdInputLocator = By.xpath("//ul[contains(@class, 'AccountFrm_country_id')]/parent::*");
+    public By loginNameInputLocator = By.id("AccountFrm_loginname");
+    public By passwordInputLocator = By.id("AccountFrm_password");
+    public By passConfirmInputLocator = By.id("AccountFrm_confirm");
+    public By newsletterInputLocator = By.xpath("//*[@id='AccountFrm_newsletter1']/button");
```
Comment (dariabajda) [inline src/main/java/pages/GuestCheckoutPage.java:28]: I always prefer to use more relative paths in locators: `//*[@id='AccountFrm_newsletter1']//button` 
That way someone could introduce e.g. `div` in the middle and test still won't fail :)

File: src/main/java/pages/GuestCheckoutPage.java (hunk 2 above)
Comment (dariabajda) [reply src/main/java/pages/GuestCheckoutPage.java:35]: https://resources.jetbrains.com/storage/products/intellij-idea/docs/IntelliJIDEA_ReferenceCard.pdf
Reformat code & Optimize imports shortcuts are super useful!

File: src/test/java/temporary/tests/TestNewCustomer.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
+package temporary.tests;
+
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import pages.LoginPage;
+
+import static engine.drivers.WebDriverFactory.quitWebDriver;
+import static org.assertj.core.api.Assertions.assertThat;
+import static steps.MainPageSteps.*;
```
Comment (dariabajda) [inline src/test/java/temporary/tests/TestNewCustomer.java:10]: It is a good practice not to use wildcards imports.
How to disable it in IntelliJ: https://www.baeldung.com/intellij-disable-wildcard-import

File: src/test/java/temporary/tests/TestNewCustomer.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    public void tearDown() {
+        quitWebDriver();
+    }
+
+    @Test
+    public void continueAsGuest() {
+        // Given:
+        openPage();
+        addToCart();
+        goToCart();
```
Comment (dariabajda) [inline src/test/java/temporary/tests/TestNewCustomer.java:30]: Why twice?

File: src/main/java/pages/GuestCheckoutPage.java (hunk 2 above)
Comment (ilku12) [reply src/main/java/pages/GuestCheckoutPage.java]: done

File: src/main/java/pages/GuestCheckoutPage.java (hunk 3 above)
Comment (ilku12) [reply src/main/java/pages/GuestCheckoutPage.java]: Address_1 and Address_2 are two separate ones. CamelCase - done.

File: src/main/java/pages/GuestCheckoutPage.java (hunk 4 above)
Comment (ilku12) [reply src/main/java/pages/GuestCheckoutPage.java]: GuestCheckoutPage has been rebuilt.

File: src/main/java/pages/GuestCheckoutPage.java (hunk 5 above)
Comment (ilku12) [reply src/main/java/pages/GuestCheckoutPage.java]: GuestCheckoutPage has been rebuilt.

File: src/main/java/pages/GuestCheckoutPage.java (hunk 6 above)
Comment (ilku12) [reply src/main/java/pages/GuestCheckoutPage.java:5]: resolved

File: src/test/java/temporary/tests/TestNewCustomer.java (hunk 9 above)
Comment (ilku12) [reply src/test/java/temporary/tests/TestNewCustomer.java]: In my branch the test is unnecessary, deleted

File: src/test/java/temporary/tests/TestNewCustomer.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+        loginPage = new LoginPage();
+    }
+
+    @After
+    public void tearDown() {
+        quitWebDriver();
+    }
+
+    @Test
+    public void continueAsGuest() {
```
[the remark this answers is not in the document]
Comment (ilku12) [reply src/test/java/temporary/tests/TestNewCustomer.java]: resolved

**Review by jakub-krzysiek — APPROVED** [review]
Approved :)

File: src/main/java/engine/property/manager/PropertyManager.java (hunk 1 above)
Comment (ilku12) [reply src/main/java/engine/property/manager/PropertyManager.java:47]: resolved

File: src/main/java/pages/GuestCheckoutPage.java (hunk 8 above)
Comment (ilku12) [reply src/main/java/pages/GuestCheckoutPage.java]: changed

File: src/test/java/temporary/tests/TestOpenPageChrome.java (hunk 7 above)
Comment (ilku12) [reply src/test/java/temporary/tests/TestOpenPageChrome.java:28]: resolved

**Pull request merged**